sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+
+from ..utils.validation import check_is_fitted
+from ..utils.validation import column_or_1d
+from ._encoders import _BaseEncoder
+
+
+class TargetRegressorEncoder(_BaseEncoder):
+    """Target Encoder for Regression Targets.
+
+    Each category is encoded based its effect on the target variable. The
+    encoding scheme takes a weighted average estimated by a multilevel
+    linear model.
+
+    Read more in the :ref:`User Guide <target_regressor_encoder>`.
+
+    Parameters
+    ----------
+    categories : 'auto' or a list of array-like, default='auto'
+        Categories (unique values) per feature:
+
+        - 'auto' : Determine categories automatically from the training data.
+        - list : `categories[i]` holds the categories expected in the ith
+          column. The passed categories should not mix strings and numeric
+          values within a single feature, and should be sorted in case of
+          numeric values.
+
+        The used categories can be found in the `categories_` attribute.
+
+    Attributes
+    ----------
+    cat_encodings_ : list of ndarray
+        The encoding corresponding to the categories in the `categories_`
+        attribute.
+
+    categories_ : list of ndarray
+        The categories of each feature determined during fitting
+        (in order of the features in X and corresponding with the output
+        of :meth:`transform`).
+
+    y_mean_ : float
```

Comment:
why expose the mean and not the variance? Or the conditional means/ variances?

I feel like we should either expose all, or none. I'd be in favor of not exposing anything.

Reply:
The mean is used for encoding unknown or missing values.

Reply:
I don't feel like this is enough of a reason to expose it.

If we choose to expose it, the docstring should say "... used for encoding unknown or missing values"